### ISSUE TITLE
Add the word `Graceful` to the reboot page

### ISSUE THREAD

**Issue by shoulders** [webmin/webmin#2119]
|  SYSTEM INFORMATION||
|----------------------|---------------------------|
|  OS type and version |     Ubuntu Linux 22.04.3     |
|  Webmin version      |     2.105      |
|  Usermin version     |     2.005     |
|  Virtualmin version  |     7.10.0       |
|  Theme version       |     21.09.5      |
|  Package updates     |     All installed packages are up to date   |

Webmin --> System --> Bootup and Shutdown --> Reboot system

![image](https://github.com/webmin/webmin/assets/319997/a5a5a3e4-eb3f-4007-9d00-8181d261267a)

Can the word `Graceful` to the description as shown above. the reason I ask for this is that I come from cPanel/WHM where there are 2 types of server reboot, Graceful and forced.

Adding this might be a little re-assuring to the newbies (i.e. I was one of these)

**Example**

> Are you sure you want to perform a graceful reboot of the system with the reboot command ?

**Comment by jcameron** [comment]
I suppose we could add this, although I feel like it would only make sense if there was also a force reboot button.

**Comment by shoulders** [comment]
The type of reboot should still be identified, I had to make the assumption it was graceful.

**Comment by iliaross** [comment]
> The type of reboot should still be identified even if just one type, I had to make the assumption it was graceful.

No, it's not really going to help because the command is configurable in the module config, and can be just about anything, so on "Reboot" page we just say which command is used, and the button text uses general text without any specifics.

**Comment by shoulders** [comment]
It should not be configurable. You should just offer a graceful reboot and a forced reboot with an appropriate bit of text explaing the 2 options.

This is a key feature and those of us who use a GUI require assurance we are doing the right thing. If I could do everything through the command line with multiple switches I would not use webmin/virtualmin. 😄

**Comment by jcameron** [comment]
I would actually be in favor of adding a separate "force reboot" button.

**Comment by shoulders** [comment]
- I am in favour of 2 buttons, `Graceful Reboot` and a `Force Reboot`.
- I know @iliaross mentioned that the buttons actual command can be configured in the settings, if this is needed maybe a 3rd button or push admins to use the CLI?
- Are 2 buttons needed for shutdown?

I think this was some additional code to aid a graceful shutdown

- https://github.com/virtualmin/virtualmin-gpl/pull/1122
- from https://forum.virtualmin.com/t/backup-issues-after-latest-virtualmin-webmin-update/134960/129

**Comment by jcameron** [comment]
Ok, I'll work on adding separate graceful and forced reboot buttons.

**Comment by shoulders** [comment]
Good stuff.

**Comment by iliaross** [comment]
> * and a `Force Reboot`.

What is the point of having force reboot!? You never need to use force reboot; the whole idea doesn't make sense. If you want force reboot pull the plug out, or tell hypervisor to stop.

We don't need another confusing option for users to choose between. Besides, rebooting must always be graceful!

**Comment by jcameron** [comment]
I've seen situations where a regular reboot won't complete because some shutdown script is hung. A force reboot will at least properly unmount filesystems, which is not the case if you just pull the plug.

**Comment by iliaross** [comment]
LOLL!! 

> I've seen situations where a regular reboot won't complete because some shutdown script is hung.

None of these can be solved by Webmin in practice! Webmin cannot reliably act as an "escalation console" after shutdown has started! `webmin.service` is ordered `After=network.target` and on shutdown it will be stopped before `network.target` is stopped in inverse order. Webmin is **often among the earlier** user-facing network services to go down during a reboot, as, again, the shutdown order is the inverse of the startup order based on those dependencies!

And, it's best not to interrupt the process and to let it finish gracefully, as tearing it down, may and most likely **will cause data loss**. If things get stuck, a user should investigate why the process is hanging or taking a long time and resolve those issues. If the reboot is truly stuck, Webmin will not be able to help!

Sorry, I cannot justify adding a forceful reboot button, because it shouldn't exist in the first place. If we were a hypervisor, then shutdown, reboot, and stop (force-stop) buttons would make sense, but we're not. We're just one of the processes on the system, not the system itself.

However, I’m open to reconsider if there are rock-solid reasons for it to exist, taking into account the points above.

**Comment by jcameron** [comment]
You're assuming `systemd` is in use - but other boot systems exist, and they might have a different shutdown order. I have absolutely seen shutdowns blocked due to Squid being really slow to exit for example.

**Comment by iliaross** [comment]
Yes, I agree. I don't doubt your experience. And, I also know that in particular the shutdown process for Squid can be slow, including regular service restarts, even on `systemd` systems. Still, it's all configurable, and we, again, shouldn't use a sledgehammer to solve a delicate problem.

However, we won't solve the problem nicely here for the reasons I mentioned earlier. The user only brought this up as a feature request for _systemd_ systems. By the way, _systemd_ has become overwhelmingly dominant in the past 10 years, with about 95%-99% of all Linux systems using it.

Ideally, an admin should know which services take a long time to shut down, like Squid or a busy Apache server, and handle them properly before rebooting to reduce downtime.
